transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -190,12 +189,16 @@ internal class MessagingClientImpl(
         }
     }
 
+    @Throws(IllegalStateException::class, IllegalArgumentException::class)
     override fun attach(
         byteArray: ByteArray,
         fileName: String,
         uploadProgress: ((Float) -> Unit)?,
     ): String {
         log.i { "attach(fileName = $fileName)" }
+        if (!attachmentHandler.validate(byteArray)) {
```

Comment:
I don't think it's bad that this code is here, but I'm wondering if it can be moved to the `prepare()` function. Do you need a public `validate()` function for use in other places?

Maybe moving this into the attachment handler will make the file preparation process more encapsulated and minimize the amount of code in the Messaging client.

Reply:
hmmm...totally make sense! Will move it to `prepare`